dissect/target/loaders/kape.py

```diff
@@ -15,25 +18,64 @@
 USNJRNL_PATHS = ["$Extend/$J", "$Extend/$UsnJrnl$J"]
 
 
+def open_vhdx(path: Path) -> Iterator[Filesystem]:
+    container = VhdxContainer(path)
+    volume_system = volume.open(container)
+    for vol in volume_system.volumes:
+        yield filesystem.open(vol)
+
+
+def is_drive_letter(path: str) -> bool:
+    return len(path) == 1 or (len(path) == 2 and path[1] == ":")
+
+
+def is_valid_kape_dir(path: Path) -> bool:
+    os_type, dirs = find_dirs(path)
+    if os_type == OperatingSystem.WINDOWS:
+        for dir_path in dirs:
+            for path in USNJRNL_PATHS:
+                if dir_path.joinpath(path).exists():
+                    return True
+
+    return False
+
+
+def is_valid_kape_vhdx(path: Path) -> bool:
+    if path.suffix == ".vhdx":
+        for fs in open_vhdx(path):
+            return is_valid_kape_dir(fs.path())
+
+    return False
+
+
 class KapeLoader(DirLoader):
     """Load KAPE forensic image format files.
 
     References:
         - https://www.kroll.com/en/insights/publications/cyber/kroll-artifact-parser-extractor-kape
     """
 
+    def __init__(self, path, **kwargs):
+        super().__init__(path)
+
+        if path.is_dir():
+            self.map = self.map_dir
+        if path.suffix == ".vhdx":
+            self.map = self.map_vhdx
+
```

Comment:
```suggestion
```
Ideally just do this in code in the `.map()` method instead of swapping out methods.